prymer/api/picking.py

```diff
@@ -136,42 +139,60 @@ def build_primer_pairs(
         region_end = max(p.span.end for p in right_primers)
         bases = fasta.fetch(target.refname, region_start - 1, region_end)
 
+    # Each tuple is left_idx, right_idx, penalty, tm
+    pairings: list[Tuple[int, int, float, float]] = []
+
+    for i in range(0, len(left_primers)):
+        for j in range(0, len(right_primers)):
+            lp = left_primers[i]
+            rp = right_primers[j]
```

Comment:
**suggestion** You could save a few lines and a layer of indentation in the remaining body by using `itertools.product`

```suggestion
    for lp, rp in itertools.product(left_primers, right_primers):
```